### DISCUSSION TITLE
Is there any way to incrementally regenerate the wp-search.json file?

### DISCUSSION THREAD

**Discussion by spencersmb** [colbyfayock/next-wordpress-starter#470] [Q&A]
I have a fairly large blog and was considering going next.js / headless and wondering if I would need to rebuild each time a new post gets released weekly.

Great work on this repo!

**Comment by colbyfayock** [comment]
hey @spencersmb i suspect this question came before the video of mine you found: https://www.youtube.com/watch?v=_isECpf3-8s

that's certainly an option (and currently what im doing on my site as i just haven't implemented it yet (procrastinating))

but ISR is probably going to be your friend here: https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation

you can create an endpoint that serves as the endpoint WordPress would hit as part of a webhook

you would need to be able to pass in the route you want to refresh or have a way of looking it up

**Comment by spencersmb** [comment]
@colbyfayock - Really appreciate the help man. I think I ended up coming up with a decent solution. I'd be curious to get your thoughts on it. Basically I have a blog with over 600+ posts that I want to use with next, but I don't want to have to rebuild each week on a new blog post. Cue your awesome video to help solve that problem! So I still want super fast search using my wp-search.json file but again, don't want to have to rebuild the whole site each week. 

Solution: 
Host a wp-search.json file in aws cloudfront for CDN, that loads in the background when someone visits the site. Then cache it in global state so it's there whenever someone wants to use search.

Each week when a new post publishes, I have an action in wordpress that fires only when a post publishes. This makes an api call in PHP to an API route on my next website. That Api Route then fetches the latest data from my backend, and saves/overwrites the json file on my S3 bucket. AWS will then clear the cache with an updated file.

And now we don't have to rebuild each week. What do you think?

**Reply by colbyfayock** [reply]
feels a bit complicated for the need but dont have a better solution off the top of my head, could be a nice option!

you could alternatively think about using edge functions with a cache header:

https://vercel.com/docs/concepts/functions/edge-functions/edge-caching

**Reply by spencersmb** [reply]
Oh its definitely a little complex lol. But I just haven't found a simple solution other than rebuilding the site each week and I've been bouncing between Next.js and Remix for awhile now. I guess I've always looked down on rebuilding each week for some reason. CMS updates should change content not the entire website, it feels inefficient to me. Maybe thats just the new way of doing things? I don't know, but I do appreciate your time and input you've given me @colbyfayock. You're a fantastic teacher and developer to learn from, so thank you!
